projectDocs/dev/developerGuide/developerGuide.t2t

```diff
@@ -781,24 +781,44 @@ This manifest file contains key = value pairs declaring info such as the add-on'
 Although it is highly suggested that manifests contain all fields, the fields marked as mandatory must be included.
 Otherwise, the add-on will not install.
 
-- name (string): A short unique name for the add-on. This is used to differentiate add-ons internally and is not shown to the user. (Mandatory)
-- summary (string): The name of the add-on as shown to the user. (Mandatory)
-- version (string): The version of this add-on; e.g. 2.0. (Mandatory)
-- author (string): The author of this add-on, preferably in the form Full Name <email address>; e.g. Michael Curran <mick@kulgan.net>. (Mandatory)
+- name (string, required): A short, unique, name for the add-on.
+  - The recommended convention is lowerCamelCase.
+  - This is used to differentiate add-ons internally, and is also used as the name of the add-on's directory in the user configuration directory.
+  - Special characters should be avoided as the add-on name will be used as a folder name.
+  Expected characters are alphanumeric, space, underscore and hyphen.
+  -
+- summary (string, required): The name of the add-on as shown to the user.
+- version (string, required): The version of this add-on; e.g. 2.0.1.
+When uploading to the Add-on Store certain requirements apply:
+  - Using ``<major>.<minor>`` or ``<major>.<minor>.<patch>`` format.
+  - For a user to be able to update to this add-on, the version must be greater than the last version uploaded.
+  - Add-on versions are expected to be unique for the addon name and channel, meaning that a beta, stable and dev version of the same add-on cannot share a version number.
+  This is so there can be a unique ordering of newest to oldest.
+  - The suggested convention is to increment the patch version number for dev versions, increment the minor version number for beta versions, and increment the major version number for stable versions.
```

Comment:
Sorry, still a remark; let me know if it should be discussed out of this PR though.

The suggested add-on versionning is still unclear to me.

What versions would you suggest for the following releases, all before 2024.1 stable release?
1. First stable release of MyAddon, compatible with 2023.3 but not with 2024.1
2. Dev release of MyAddon implementing compatibility with 2024.1 (including incompatible changes)
3. Stable release on top of the first stable, containing a hotfix (e.g. due to external change such as new Windows update)
4. Dev release on top of the last dev, containing the same hotfix.

Reply:
If you don't want to follow the conventions exactly, this should be fine, particularly if (2) is a major update to the add-on: 1.0.0, 2.0.0, 1.0.1, 2.0.1

However, I think the convention ordering might still be fine/better: 1.0.0, 1.0.1, 2.0.1, 2.0.2

This ensures:
- (2) will not update to (3) without a user overriding compatibility
- (2) and (4) won't show up to users on 2023.3 if they have a minimum API version of 2024.1
- If all 4 releases have cross 2023.3/2024.1 compatibility, a user can be encouraged to update to each sequential version (e.g. (3) if (4) takes a while to be released).

Reply:
Maybe this could be improved by saying version numbers can also be arbitrarily increased for "base" major addon update releases. e.g 1.x.x dev/beta/stable becomes 2.0.0 stable

Reply:
Also note that we will be able to abandon this convention for a simpler one, if we can include a patch number and/or a separate version string

Reply:
@cyrilleB79, I try to follow this scheme in all my add-ons. Hope it"s well followed, unless some mistake from my part. You may search nvdaes in the store to see some examples, and if something is wrong I"ll try to fix it in the future, since my intention was allways to follow this convention for versions and channels.